packages/nc-gui-v2/components/smartsheet/expanded-form/index.vue

```diff
@@ -0,0 +1,136 @@
+<script setup lang="ts">
+import type { ColumnType, TableType } from 'nocodb-sdk'
+import { isVirtualCol } from 'nocodb-sdk'
+import { useVModel } from '@vueuse/core'
+import { computed, provide, toRef, watch } from 'vue'
+import Comments from './Comments.vue'
+import Header from './Header.vue'
+import { NOCO } from '~/lib'
+import { extractPkFromRow } from '~/utils'
+import { useNuxtApp } from '#app'
+import { useProvideSmartsheetStore, useSmartsheetStoreOrThrow } from '~/composables'
+import type { Row } from '~/composables'
+import { useProvideExpandedFormStore } from '~/composables/useExpandedFormStore'
+import { FieldsInj, IsFormInj, MetaInj } from '~/context'
+
+interface Props {
+  modelValue: string | null
+  row: Row
+  state?: Record<string, any> | null
+  meta: TableType
+  loadRow?: boolean
+
+  useMetaFields?: boolean
+}
+
+const props = defineProps<Props>()
+const emits = defineEmits(['update:modelValue'])
+const _fields = inject(FieldsInj, ref([]))
+const row = toRef(props, 'row')
+const state = toRef(props, 'state')
+const meta = toRef(props, 'meta')
+
+const fields = computed(() => {
+  if (props.useMetaFields) {
+    return meta.value.columns ?? []
+  }
+  return _fields.value
+})
+
+provide(MetaInj, meta)
+
+const { commentsDrawer, changedColumns, state: rowState } = useProvideExpandedFormStore(meta, row)
+
+const { $api } = useNuxtApp()
+if (props.loadRow) {
+  const { project } = useProject()
+  row.value.row = await $api.dbTableRow.read(
+    NOCO,
+    project.value.id as string,
+    meta.value.title,
+    extractPkFromRow(row.value.row, meta.value.columns as ColumnType[]),
+  )
+  row.value.oldRow = { ...row.value.row }
+  row.value.rowMeta = {}
+}
```

Comment:
I think in the bellow scenario, it will 

```vue
   <SmartsheetExpandedForm
      v-if="expandedFormDlg && expandedFormRow"
      v-model="expandedFormDlg"
      :row="{ row: expandedFormRow }"
      :meta="relatedTableMeta"
      load-row
      use-meta-fields
    />
```

Reply:
yeah, like I said, if it's initially true, it will do it (but then the question is, why would the prop be necessary at all if it's just going to be true anyway?) or is it just supposed to be true or false initially and never change?
If that's the case then it's ok I guess 😄